Classes/git/PBGitGraphLine.h

```diff
@@ -8,8 +8,8 @@
 
 struct PBGitGraphLine
 {
-	int upper      : 1;
-	int from       : 8;
-	int to         : 8;
-	int colorIndex : 8;
+	int upper       : 1;
+	long from       : 8;
+	long to         : 8;
+	int colorIndex  : 8;
```

Comment:
To be honest, it’s unclear to me what should be our goal here and how to best achieve it in the language. To me it feels like either the code is broken or the types aren’t correct.

Would solving this at `add_line` involve sth à la

```
int long_to_integer_heuristic(long input)
{
	if (input < INT_MIN) {
		return INT_MAX;
	} else if (input > INT_MAX) {
		return INT_MIN;
	}
	return (int)input;
}
```
?